core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -338,6 +338,11 @@ public void sendMessage(ReqT message) {
   }
 
   @Override
+  public void setMessageCompression(boolean enabled) {
```

Comment:
It's better to use `Preconditions.checkState()` to give the caller a more informative exception if this is called before `start()`.